test/test_swmm.py

```diff
@@ -81,7 +81,7 @@ def test_count_vw_vertices(self):
         self.assert_count("vw_vertices", "qgep_swmm", 32)
 
     def test_count_vw_xsections(self):
-        self.assert_count("vw_xsections", "qgep_swmm", 2348)
+        self.assert_count("vw_xsections", "qgep_swmm", 17200)
```

Comment:
Is this change of result expected from the above changes ?

From the code, it seems to indicate that `ws._function_hierarchic` is not the same as `ch.function_hierarchic`, which seems surprising, no ? Could it indicate some other issue, e.g. a trigger that doesn't properly update it ?

Reply:
Indeed if you run: `
SELECT DISTINCT
  re.obj_id as Link,
  ws._function_hierarchic,
  ch.function_hierarchic
FROM qgep_od.reach re
LEFT JOIN qgep_od.pipe_profile pp on pp.obj_id = re.fk_pipe_profile
LEFT JOIN qgep_od.wastewater_networkelement ne ON ne.obj_id::text = re.obj_id::text
LEFT JOIN qgep_od.wastewater_structure ws ON ws.obj_id::text = ne.fk_wastewater_structure::text
LEFT JOIN qgep_od.channel ch ON ch.obj_id::text = ws.obj_id::text
WHERE ch.function_hierarchic = ANY (ARRAY[5066, 5068, 5069, 5070, 5064, 5071, 5062, 5072, 5074]);
`
The two columns are not similar. I think that the trigger that computed ws._function_hierarchic was mainly thought for the point-like objects (manhole etc.)

Reply:
> The two columns are not similar. I think that the trigger that computed ws._function_hierarchic was mainly thought for the point-like objects (manhole etc.)

That's right! The two columns can be different.

Reply:
So yes the change of result is expected from the above change.